src/main/scala/org/scalajs/cli/Scalajsld.scala

```diff
@@ -170,27 +203,53 @@ object Scalajsld {
 
       val linker = StandardImpl.linker(config)
       val logger = new ScalaConsoleLogger(options.logLevel)
-
-      val output = {
-        val js = options.output.toPath()
-        val sm = js.resolveSibling(js.getFileName().toString() + ".map")
-
-        def relURI(f: Path) =
-          new URI(null, null, f.getFileName().toString, null)
-
-        LinkerOutput(PathOutputFile(js))
-          .withSourceMap(PathOutputFile(sm))
-          .withSourceMapURI(relURI(sm))
-          .withJSFileURI(relURI(js))
-      }
-
       val cache = StandardImpl.irFileCache().newCache
 
       val result = PathIRContainer
         .fromClasspath(classpath)
         .flatMap(containers => cache.cached(containers._1))
-        .flatMap(linker.link(_, moduleInitializers, output, logger))
+        .flatMap{ irFiles =>
+          (options.output, options.outputDir) match {
+            case (Some(jsFile), None) =>
+              DeprecatedLinkerAPI().link(linker, irFiles.toList, moduleInitializers, jsFile, logger)
+            case (None, Some(outputDir)) =>
+              linker.link(irFiles, moduleInitializers, PathOutputDirectory(outputDir.toPath), logger)
+            case _ => throw new Exception("Either output or outputDir have to be defined.")
+          }
+        }
       Await.result(result, Duration.Inf)
     }
   }
+
+  // Covers deprecated api with not deprecated method. Suppresses warning.
+  private abstract class DeprecatedLinkerAPI {
+    def link(linker: Linker,
+             irFiles: Seq[IRFile],
+             moduleInitializers: Seq[ModuleInitializer],
+             linkerOutputFile: File,
+             logger: Logger): Future[Unit]
```

Comment:
The indentation here is not following SJS style: https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#indentation

Either binpack it (as many args per line as possible, 4 spaces indent), or one per line, 2 spaces indented.

Reply:
I'm not sure, if it's this case. It'd be right for function invocation, but no declaration. I applied rule `In general, indentation is 2 spaces, except continuation lines, which are indented 4 spaces.` following example `org.scalajs.linker.interface.Linker#link`